### PR TITLE
Add flag for outputting IRs

### LINKED ISSUE #303: Flag for outputting IRs
Following the recently merged (#294) flags `--build` and `--compile`, I would really appreciate having a compiler flag for outputting the generated intermediate representations (as is done [here on the website](https://effekt-lang.org/docs/implementation/lift-inference)).

### PR DESCRIPTION
**Pull request by marvinborner**
This closes #303.

It adds two new flags:

- `showIR=<stage>`, prints the selected IR to stdout
- `writeIRs`, writes all IRs to the output directory

### CODE REVIEW

File: effekt/shared/src/main/scala/effekt/generator/llvm/LLVM.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ class LLVM extends Compiler[String] {
   // -----------------------------------
   object steps {
     // intermediate steps for VSCode
-    val afterCore = allToCore(Core) map { c => c.main }
+    val afterCore = allToCore(Core) andThen Aggregate andThen core.Optimizer
```
Comment (marvinborner) [inline effekt/shared/src/main/scala/effekt/generator/llvm/LLVM.scala:56]: I'm a bit unsure about this. While it fixes the issue, I can't use `map { c => c.main }` anymore. Also, was there any reason to exclude the aggregation and optimization phases?

Comment (b-studios) [reply effekt/shared/src/main/scala/effekt/generator/llvm/LLVM.scala:56]: Can you please also add polymorphism boxing?

File: effekt/jvm/src/main/scala/effekt/Driver.scala [hunk 2]
```diff
@@ -115,6 +117,20 @@ trait Driver extends kiama.util.Compiler[EffektConfig, EffektError] { outer =>
     }
   }
 
+  def showIR(source: Source, config: EffektConfig)(implicit C: Context): Unit =
+    config.showIR().map { stage =>
+      C.compiler.prettyIR(source, stage).map { case (Document(s, _)) => println(s) }
```
[the remark this answers is not in the document]
Comment (marvinborner) [reply effekt/jvm/src/main/scala/effekt/Driver.scala:122]: I'm not sure. I mostly decided to use `println` so you could use Unix pipes. Although this isn't all that useful anyway if the program also returns other output. Maybe the IRs should be printed to `stderr`?

[the remark this answers is not in the document]
Comment (b-studios) [reply effekt/jvm/src/main/scala/effekt/Driver.scala:122]: stderr would be the wrong semantics IMO.

I guess println is fine for now.

**Pull request merged**